cli/src/pcluster/templates/imagebuilder_stack.py

```diff
@@ -0,0 +1,177 @@
+        core.CfnParameter(
+            self,
+            "UpdateAndReboot",
+            type="String",
+            default=str.lower(str(dev_settings.update_os_and_reboot)),
+            description="UpdateAndReboot",
+        )
+
+        ami_info = utils.get_info_for_amis([build.parent_image])
+        architecture = ami_info[0].get("Architecture")
+        if build.instance_type:
+            instance_types = [build.instance_type]
+        else:
+            instance_types = ["p2.xlarge", "c5.xlarge"] if architecture == "x86_64" else ["m6g.xlarge"]
+
+        # Setup ImageBuilder Resources
+        resources_prefix = utils.generate_random_prefix()
+
+        # InstanceRole
+        if build.instance_role:
+            managed_policy_arns = [build.instance_role]
+        else:
+            managed_policy_arns = [
+                core.Fn.sub("arn:${AWS::Partition}:iam::aws:policy/AmazonSSMManagedInstanceCore"),
+                core.Fn.sub("arn:${AWS::Partition}:iam::aws:policy/EC2InstanceProfileForImageBuilder"),
+            ]
+        instancerole = iam.CfnRole(
+            self,
+            id="InstanceRole",
+            managed_policy_arns=managed_policy_arns,
+            assume_role_policy_document={
+                "Statement": {
+                    "Action": "sts:AssumeRole",
+                    "Effect": "Allow",
+                    "Principal": {"Service": "ec2.amazonaws.com"},
+                },
+                "Version": "2012-10-17",
+            },
+            path="/executionServiceEC2Role/",
+        )
+        instancerole.add_metadata("Comment", "Role to be used by instance during image build.")
+
+        # InstanceProfile
+        iam.CfnInstanceProfile(
+            self, id="InstanceProfile", path="/executionServiceEC2Role/", roles=[core.Fn.ref("InstanceRole")]
+        )
+
+        # InfrastructureConfiguration
+        imagebuilder.CfnInfrastructureConfiguration(
+            self,
+            id="PClusterImageInfrastructureConfiguration",
+            name="-".join(["PCluster-Image-Infrastructure-Configuration", resources_prefix]),
+            instance_profile_name=core.Fn.ref("InstanceProfile"),
+            terminate_instance_on_failure=dev_settings.terminate_instance_on_failure.b if dev_settings else True,
+            instance_types=instance_types,
+        )
+
+        # Define ami build instance ebs
+        if image.root_volume:
+            ebs = imagebuilder.CfnImageRecipe.EbsInstanceBlockDeviceSpecificationProperty(
+                volume_size=image.root_volume.size
+                or ami_info[0].get("BlockDeviceMappings")[0].get("Ebs").get("VolumeSize") + 15,
+                volume_type="gp2",
+            )
+        else:
+            ebs = imagebuilder.CfnImageRecipe.EbsInstanceBlockDeviceSpecificationProperty(
+                volume_size=ami_info[0].get("BlockDeviceMappings")[0].get("Ebs").get("VolumeSize") + 15,
```

Comment:
Let's define a constant for this [magic number](https://en.wikipedia.org/wiki/Magic_number_(programming)).

Reply:
Done